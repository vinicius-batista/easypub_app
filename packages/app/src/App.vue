<template>
  <v-app style="background-color: #f1f3f5">
    <v-container :style="{ padding: 0 }">
      <router-view></router-view>
    </v-container>
  </v-app>
</template>

<script>
export default {}
</script>
