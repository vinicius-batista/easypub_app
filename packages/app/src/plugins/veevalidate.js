import Vue from 'vue'
import VeeValidate, { Validator } from 'vee-validate'
import VeeValidateMessagesBR from 'vee-validate/dist/locale/pt_BR'

Validator.localize('pt_BR', VeeValidateMessagesBR)
Vue.use(VeeValidate, { locale: 'pt_BR' })
