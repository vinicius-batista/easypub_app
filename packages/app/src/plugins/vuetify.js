import Vue from 'vue'
import { colors } from 'vuetify/lib'
import pt from 'vuetify/es5/locale/pt'
import Vuetify from 'vuetify'

import VueTheMask from 'vue-the-mask'
Vue.use(VueTheMask)

Vue.use(Vuetify)

export default new Vuetify({
  lang: {
    locales: { pt },
    current: 'pt',
  },
  icons: {
    iconfont: 'fa',
  },
  theme: {
    themes: {
      light: {
        primary: colors.blue.accent4,
        secondary: colors.lightBlue.lighten2,
        tertiary: colors.blueGrey.darken2,
      },
    },
  },
})
