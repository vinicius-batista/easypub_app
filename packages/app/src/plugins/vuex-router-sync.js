import { sync } from 'vuex-router-sync'

export default (store, router) => sync(store, router)
