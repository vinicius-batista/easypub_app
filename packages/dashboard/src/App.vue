<template>
  <v-app style="background-color: #f1f3f5">
    <router-view></router-view>
  </v-app>
</template>

<script>
export default {
  name: 'App',
}
</script>
