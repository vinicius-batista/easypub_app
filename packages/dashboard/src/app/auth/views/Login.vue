<template>
  <AuthForm
    buttonText="Login"
    helperText="NÃ£o possui uma conta?"
    actionText="Cadastre-se agora!"
    @action-text:clicked="changeRoute('auth.register')"
    data-testId="login-page"
  >
    <ValidationProvider
      v-for="{ label, model, icon, type, rules } in form"
      :key="label"
      v-bind="{ rules }"
      :name="label"
    >
      <template v-slot="{ errors }">
        <v-text-field
          class="my-2"
          :type="type"
          :label="label"
          v-model="input[model]"
          :prepend-icon="icon"
          :error-messages="errors"
          filled
          :data-testId="model"
        />
      </template>
    </ValidationProvider>
  </AuthForm>
</template>

<script>
import AuthForm from '../components/AuthForm'
import { ValidationProvider } from 'vee-validate'

export default {
  name: 'Login',
  components: { AuthForm, ValidationProvider },
  data: () => ({
    input: {
      email: '',
      password: '',
    },
    form: [
      {
        label: 'Email',
        model: 'email',
        icon: 'fa-envelope',
        rules: 'required|email',
      },
      {
        label: 'Senha',
        model: 'password',
        icon: 'fas fa-lock',
        type: 'password',
        rules: 'required|min:6',
      },
    ],
  }),
  methods: {
    changeRoute(name) {
      this.$router.push({ name })
    },
  },
}
</script>
