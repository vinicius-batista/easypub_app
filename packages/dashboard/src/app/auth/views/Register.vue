<template>
  <AuthForm
    buttonText="Cadastre-se"
    helperText="Já possui uma conta?"
    actionText="Entre agora!"
    data-testId="register-page"
    @action-text:clicked="changeRoute('auth.login')"
  >
    <ValidationProvider
      v-for="{ label, icon, type, rules, model } in form"
      :key="label"
      v-bind="{ rules }"
      :name="label"
    >
      <template v-slot="{ errors }">
        <v-text-field
          class="my-2"
          :type="type"
          :label="label"
          v-model="input[model]"
          :prepend-icon="icon"
          :error-messages="errors"
          filled
          :data-testId="model"
        />
      </template>
    </ValidationProvider>
    <ValidationProvider rules="required|size:2000000" name="Foto do bar">
      <template v-slot="{ errors }">
        <v-file-input
          class="my-2"
          label="Foto do bar"
          accept="image/png, image/jpeg, image/bmp"
          v-model="input['barAvatar']"
          prepend-icon="fa-image"
          :error-messages="errors"
          filled
          data-testId="barAvatar"
        />
      </template>
    </ValidationProvider>
  </AuthForm>
</template>

<script>
import AuthForm from '../components/AuthForm'
import { ValidationProvider } from 'vee-validate'

export default {
  name: 'Register',
  components: { AuthForm, ValidationProvider },
  data: () => ({
    input: {
      email: '',
      password: '',
      phone: '',
      name: '',
      barName: '',
      barAddress: '',
      barAvatar: null,
    },
    form: [
      {
        label: 'Nome',
        model: 'name',
        icon: 'fa-user',
        rules: 'required|alpha_spaces',
      },
      {
        label: 'Email',
        model: 'email',
        icon: 'fa-envelope',
        rules: 'required|email',
      },
      {
        label: 'Telefone',
        model: 'phone',
        icon: 'fa-phone',
        rules: 'required',
        mask: '(##) ##### - ####',
        type: 'tel',
      },
      {
        label: 'Senha',
        model: 'password',
        icon: 'fas fa-lock',
        type: 'password',
        rules: 'required|min:6',
      },
      {
        label: 'Nome do bar',
        model: 'barName',
        icon: 'fa-beer',
        rules: 'required',
      },
      {
        label: 'Endereço do bar',
        model: 'barAddress',
        icon: 'fas fa-map-marker-alt',
        rules: 'required',
      },
    ],
  }),
  methods: {
    changeRoute(name) {
      this.$router.push({ name })
    },
  },
}
</script>
