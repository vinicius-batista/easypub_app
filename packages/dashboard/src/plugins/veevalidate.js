import { localize, extend } from 'vee-validate'
import pt_BR from 'vee-validate/dist/locale/pt_BR.json'
import {
  required,
  email,
  min,
  alpha_spaces,
  size,
} from 'vee-validate/dist/rules'

extend('required', required)
extend('email', email)
extend('min', min)
extend('alpha_spaces', alpha_spaces)
extend('size', size)

localize('pt_BR', pt_BR)
