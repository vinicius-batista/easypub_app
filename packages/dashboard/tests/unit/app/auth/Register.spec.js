import Vue from 'vue'
import Vuetify from 'vuetify'
Vue.use(Vuetify)

import { render, fireEvent, cleanup } from '@testing-library/vue'
import Login from '@/app/auth/views/Login'
import Register from '@/app/auth/views/Register'
import App from '@/App'
import '../../plugins/veevalidate'

const routes = [
  { path: '/auth/register', name: 'auth.register', component: Register },
  { path: '/auth/login', name: 'auth.login', component: Login },
]

describe('Register.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(cleanup)

  test('provide incorrect inputs should show error messages', async () => {
    const { getByTestId, getByText, html } = render(Register, {
      vuetify,
      sync: false,
    })

    const emailInput = getByTestId('email')
    await fireEvent.update(emailInput, 'jose.com')

    const passwordInput = getByTestId('password')
    await fireEvent.update(passwordInput, '12345')

    const nameInput = getByTestId('name')
    await fireEvent.update(nameInput, 'jose ronaldinho123')

    const barAddressInput = getByTestId('barAddress')
    await fireEvent.update(barAddressInput, '')

    const barNameInput = getByTestId('barName')
    await fireEvent.update(barNameInput, '')

    getByText('O campo Email deve ser um email válido')
    getByText('O campo Senha deve conter pelo menos 6 caracteres')
    getByText('O campo Nome só pode conter caracteres alfabéticos e espaços')
    getByText('O campo Endereço do bar é obrigatório')
    getByText('O campo Nome do bar é obrigatório')

    expect(html).toMatchSnapshot()
  })

  test('provide correct inputs should not show error messages', async () => {
    const { getByTestId, getByText, html } = render(Register, {
      vuetify,
      sync: false,
    })

    const emailInput = getByTestId('email')
    await fireEvent.update(emailInput, 'jose@jose.com')

    const passwordInput = getByTestId('password')
    await fireEvent.update(passwordInput, '123456')

    const nameInput = getByTestId('name')
    await fireEvent.update(nameInput, 'jose ronaldinho')

    const barAddressInput = getByTestId('barAddress')
    await fireEvent.update(barAddressInput, 'rua das couves, 123')

    const barNameInput = getByTestId('barName')
    await fireEvent.update(barNameInput, 'bar 234')

    expect(() =>
      getByText('O campo Email deve ser um email válido')
    ).toThrowError()

    expect(() =>
      getByText('O campo Senha deve conter pelo menos 6 caracteres')
    ).toThrowError()

    expect(() =>
      getByText('O campo Nome só pode conter caracteres alfabéticos e espaços')
    ).toThrowError()

    expect(() =>
      getByText('O campo Endereço do bar é obrigatório')
    ).toThrowError()

    expect(() => getByText('O campo Nome do bar é obrigatório')).toThrowError()

    expect(html).toMatchSnapshot()
  })

  test('click action-text should change route to register', async () => {
    const { getByTestId, html } = render(
      App,
      {
        vuetify,
        routes,
      },
      (vue, store, router) => router.push('/auth/register')
    )

    const actionText = getByTestId('action-text')
    await fireEvent.click(actionText)
    getByTestId('login-page')

    expect(html).toMatchSnapshot()
  })
})
