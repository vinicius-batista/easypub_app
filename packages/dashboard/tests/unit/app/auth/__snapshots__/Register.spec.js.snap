// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue click action-text should change route to register 1`] = `[Function]`;

exports[`Register.vue provide correct inputs should not show error messages 1`] = `[Function]`;

exports[`Register.vue provide incorrect inputs should show error messages 1`] = `[Function]`;
